src/test/java/com/alkemy/ong/controller/v1/CategoryControllerIT.java

```diff
@@ -0,0 +1,242 @@
+package com.alkemy.ong.controller.v1;
+
+import com.alkemy.ong.H2Config;
+import com.alkemy.ong.SecurityConfig;
+import com.alkemy.ong.controller.ControllerConstants;
+import com.alkemy.ong.dto.CategoryDto;
+import com.alkemy.ong.utils.JsonUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static com.alkemy.ong.utils.TestUtils.matchJson;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+@SpringJUnitConfig(classes = {H2Config.class, SecurityConfig.class})
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class CategoryControllerIT {
+    @Autowired
+    MockMvc mockMvc;
+
+
+    @Test
+    @Order(1)
+    void createCategory_shouldRespond201() throws Exception {
+        for (long i = 1; i < 7; i++) {
```

Comment:
It is a great idea to reuse a method to test different use cases, but in testing we do not use control flow statements (loops and if) for this matter, we use parametrized tests instead.

Please check: https://www.baeldung.com/parameterized-tests-junit-5

Do something like that:
```java
    @DisplayName("Create Category")
    @ParameterizedTest(name = "{displayName} - [{index}] {arguments}")
    @ValueSource(strings = {"A", "B", "C"})
    void testWithParams(String letter) {
        // test
    }
```

Reply:
Corregido!